src/main/java/org/spongepowered/common/data/manipulator/mutable/common/AbstractData.java

```diff
@@ -0,0 +1,248 @@
+        // Basic retrieval of the processor
+        Optional<ValueProcessor<E, ?>> processor = SpongeDataRegistry.getInstance().getBaseValueProcessor(key);
+        // If we have a processor, well, we can retrieve the value, if not, c'est la vie.
+        return processor.isPresent() ? processor.get().getValueFromContainer(this) : Optional.<E>absent();
+    }
+
+    @Nullable
+    @Override
+    public <E> E getOrNull(Key<? extends BaseValue<E>> key) {
+        return get(key).orNull(); // Just use the provided optional
+    }
+
+    @Override
+    public <E> E getOrElse(Key<? extends BaseValue<E>> key, E defaultValue) {
+        return get(key).or(checkNotNull(defaultValue)); // Or use the optional with a default value
+    }
+
+    @Override
+    public <E, V extends BaseValue<E>> Optional<V> getValue(Key<V> key) {
+        final Optional<ValueProcessor<E, V>> optional = SpongeDataRegistry.getInstance().getValueProcessor(key);
+        return optional.isPresent() ? optional.get().getApiValueFromContainer(this) : Optional.<V>absent();
+    }
+
+    @Override
+    public boolean supports(Key<?> key) {
+        // Again, more basic getting of the processor
+        Optional<ValueProcessor<?, ?>> processor = SpongeDataRegistry.getInstance().getValueProcessor(key);
+        // If the processor is present, and if the processor says that this container supports the key!
+        return processor.isPresent() && processor.get().supports(this);
+    }
+
+    @Override
+    public boolean supports(BaseValue<?> baseValue) {
+        // Again, more basic getting of the processor
+        Optional<ValueProcessor<?, ?>> processor = SpongeDataRegistry.getInstance().getValueProcessor(baseValue.getKey());
+        // If the processor is present, and if the processor says that this container supports the key!
+        return processor.isPresent() && processor.get().supports(this);
+    }
+
+    @Override
+    public ImmutableSet<Key<?>> getKeys() {
+        // Dunno whether I should be lazy or not. I totally could be though.
+        return ImmutableSet.of();
+    }
+
+    @Override
+    public ImmutableSet<BaseValue<?>> getValues() {
+        // Again, laziness. Should I actually override this?
+        return ImmutableSet.of();
+    }
+
+    @Override
+    public int hashCode() {
+        // The only thing we can really hashcode, overrides should be supported.
+        return Objects.hashCode(this.manipulatorClass);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        final AbstractData other = (AbstractData) obj;
+        return Objects.equal(this.manipulatorClass, other.manipulatorClass);
+    }
+
+    @Override
+    public DataContainer toContainer() {
+        // Yep, empty container
+        return new MemoryDataContainer();
```

Comment:
Not likely since the `MemoryDataContainer` is mutable, making any changes to the `EMPTY` instance would likely result in a messy `EMPTY` that isn't really `EMPTY`.